benchmarks/docker_scale_test.sh

```
@@ -0,0 +1,213 @@
+#
+# Examples:
+#   ./benchmarks/docker_scale_test.sh 2g 500000   # 2GB limit, up to 500K docs
+#   ./benchmarks/docker_scale_test.sh 4g 1000000  # 4GB limit, up to 1M docs
+# =============================================================================
+
+set -euo pipefail
+
+MEM_LIMIT="${1:-2g}"
+MAX_DOCS="${2:-500000}"
+BATCH_SIZE=10000
+IMAGE="python:3.12-slim"
```

Shellcheck (0.11.0)

[warning] IMAGE appears unused. Verify use (or export if used externally).

(SC2034)